integrationTests/src/test/java/org/sonarqube/gradle/BootstrapTest.java

```diff
@@ -0,0 +1,204 @@
+    env.put("SONAR_SCANNER_ARCH", arch);
+    RunResult result = runSonarAnalysis("/java-gradle-simple", env);
+    String url = ORCHESTRATOR.getServer().getUrl() + String.format("/api/v2/analysis/jres?os=%s&arch=%s", unsupportedOS, arch);
+    String expectedLog = String.format("Error status returned by url [%s]: 400", url);
+    assertThat(result.getLog()).contains(expectedLog);
+  }
+
+  @Test
+  public void testBootstrappingUsesProvidedJre() throws Exception {
+    String javaHome = guessJavaHomeSelectedByGradle();
+    String project = "/java-gradle-simple";
+    HashMap<String, String> env = new HashMap<>();
+    env.put("SONAR_SCANNER_JAVA_EXE_PATH", javaHome + File.separator + "bin" + File.separator + "java");
+    env.put("DUMP_SYSTEM_PROPERTIES", "java.home");
+    env.put("DUMP_SENSOR_PROPERTIES", "sonar.java.jdkHome");
+    RunResult result = runSonarAnalysis(project, env);
+    assertProvisionedJreIsUsed(project, result, false);
+    assertThat(result.getLog()).contains("Using the configured java executable");
+  }
+
+  private void assertProvisionedJreIsUsed(String projectName, RunResult result, boolean shouldBeUsed) throws IOException {
+    Properties props = readDumpedProperties(projectName);
+    SoftAssertions softly = new SoftAssertions();
+    if (shouldBeUsed) {
+      softly.assertThat(result.getLog()).contains("JRE provisioning:");
+      softly.assertThat(props.getProperty("java.home"))
+        .isNotEmpty()
+        .isNotEqualTo(System.getProperty("java.home"))
+        .isNotEqualTo(guessJavaHomeSelectedByGradle())
+        .contains(".sonar" + File.separator + "cache");
+    } else {
+      softly.assertThat(result.getLog()).doesNotContain("JRE provisioning:");
+      softly.assertThat(props.getProperty("java.home"))
+        .isNotEmpty()
+        .isEqualTo(System.getProperty("java.home"))
+        .isEqualTo(guessJavaHomeSelectedByGradle())
+        .doesNotContain(".sonar" + File.separator + "cache");
+    }
+
+  }
+
+  private RunResult runSonarAnalysis(String project, Map<String, String> env, String... args) throws Exception {
+    env.put("SONAR_HOST_URL", ORCHESTRATOR.getServer().getUrl());
+    File projectBaseDir = new File(this.getClass().getResource(project).toURI());
+    String projectDir = project.startsWith("/") ? "." + project : project;
+    File tempProjectDir = new File(temp.getRoot(), projectDir);
+    if (!tempProjectDir.exists()) {
+      tempProjectDir = temp.newFolder(projectDir);
+    }
+    File outputFile = temp.newFile();
+    FileUtils.copyDirectory(projectBaseDir, tempProjectDir);
+    List<String> command = getGradlewCommand();
+    addSQToken(command, ORCHESTRATOR.getDefaultAdminToken());
+    command.addAll(Arrays.asList("sonar", "--no-daemon", "--info"));
+    command.addAll(Arrays.asList(args));
+    File exeDir = tempProjectDir;
+    ProcessBuilder pb = new ProcessBuilder(command)
+      .directory(exeDir)
+      .redirectOutput(outputFile)
+      .redirectErrorStream(true);
```

Comment:
Since we introduce Orchestrator in this PR, would it be worth using [GradleBuild](https://github.com/SonarSource/orchestrator/blob/e2cb5d2c30c0232c70939e9219c751b187d2252e/sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/GradleBuild.java#L34) rather than a subprocess?

Reply:
I tried to, but I cannot get the logs out of that way of execution, I will leave it like this for the time being

Reply:
No problem. thank your for looking into the issue